### DISCUSSION TITLE
Can puppeteer be used from a content script to automate page interactions?

### DISCUSSION THREAD

**Discussion by Wizzel1** [wxt-dev/wxt#554] [Q&A] [answered]
As far as I know this is not possible but I am very much a beginner so maybe there is a way I don't know about.

**Comment by aklinker1** [comment, marked as answer]
No, puppeteer uses the automation APIs, which require chrome be started with specific flags. I don't think there's a way to include puppeteer and not require those flags, so I don't think it's possible.

**Reply by Wizzel1** [reply]
@aklinker1 thanks. Do you know of any library that makes it easier/more efficient to interact with the DOM than the standard javascript? 

Anyways, thank you so much for this framework, it really does a lot for me. The developer experience is mindblowing!

**Reply by aklinker1** [reply]
Glad you like it!

You could try [`@tesing-library/dom`](https://testing-library.com/docs/dom-testing-library/intro). It's meant to be for unit testing, but you might be able to bundle it with your extension and use it in a content script.
